common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -936,12 +936,14 @@ static jlong NativeCrypto_EVP_PKEY_new_RSA(JNIEnv* env, jclass, jbyteArray n, jb
         return 0;
     }
 
+#if defined(CONSCRYPT_JNI_TRACE)
```

Comment:
This works, but I think it'd be simpler if, instead, we just made the value of `kWithJniTrace` available in the header. Since we probably [don't care about the linkage](https://isocpp.org/blog/2018/05/quick-q-use-of-constexpr-in-header-file), we can probably just drop trace.cc and make them all constexprs in trace.h.

Then, at this call site, the compiler will know whether `kWithJniTrace` is true or false and, if false, just delete the code for you. And, if true, it'll save itself the bother of emitting a branch.

Reply:
Thanks, done!  Savings on MacOS x64 are exactly the same as the #ifdef version (98KB, 520 format strings), and the CL is rather more elegant this way.